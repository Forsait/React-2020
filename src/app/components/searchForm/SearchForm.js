import React from 'react';

import styles from './SearchForm.module.scss';

import Button from '../common/Button';

export default class SearchForm extends React.Component {
  constructor(props) {
    super(props);
    this.submitSearch = this.submitSearch.bind(this);
    this.input = React.createRef();
  }

  submitSearch(e) {
    e.preventDefault();
    this.props.submitForm(this.input.current.value);
  }

  render() {
    return (
      <form className={styles.wrap} onSubmit={this.submitSearch}>
        <div className={styles.m_12}>
          <input ref={this.input} type="text" placeholder="Search" defaultValue={this.props.defaultVal} />
          <Button type="submit">Search</Button>
        </div>
      </form>
    );
  }
}
