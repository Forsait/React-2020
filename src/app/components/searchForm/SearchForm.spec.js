import React from 'react';
import { shallow } from 'enzyme';

import SearchForm from './SearchForm';

describe('SearchForm component', () => {

  it('should be render correctly', () => {
    const com = shallow(<SearchForm />);
    expect(com).toMatchSnapshot()
  });

  it('should benn called submitSearch', () => {
    const submitSearch = jest.fn();
    const com = shallow(<SearchForm />);
    com.instance().submitSearch = submitSearch;
    com.instance().forceUpdate();
    com.find('form').simulate('submit');
    expect(submitSearch).toHaveBeenCalled();
  });

});