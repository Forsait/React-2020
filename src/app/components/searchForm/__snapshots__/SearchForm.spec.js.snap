// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component should be render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <input
      placeholder="Search"
      type="text"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
